vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh

```diff
@@ -83,13 +83,18 @@ echo "Granting access to $disk_resource_id for 1 hour"
 # shellcheck disable=SC2102
 sas=$(az disk grant-access --ids $disk_resource_id --duration-in-seconds 3600 --query [accessSas] -o tsv)
 
-# TBD: Need to investigate why `azcopy-preview login --login-type=MSI` does not work
-echo "Setting azcopy environment variables with pool identity: $AZURE_MSI_RESOURCE_STRING"
-export AZCOPY_AUTO_LOGIN_TYPE="MSI"
-export AZCOPY_MSI_RESOURCE_STRING="$AZURE_MSI_RESOURCE_STRING"
-
 echo "Uploading $disk_resource_id to ${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd"
-azcopy-preview copy "${sas}" "${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd" --recursive=true || exit $?
+
+if [[ "${OS_TYPE}" == "Linux" ]]; then
```

Comment:
[Just for my learning] Not fully understanding the if condition here. If it isn't a Linux, what will it be? windows?

Reply:
Yes its either Linux or windows. The windows pipeline still uses an older container that has to use the old version of azcopy.